File: src/main/kotlin/gbl/tag/type/GblProgLzma.kt
package gbl.tag.type

import gbl.tag.GblType
import gbl.tag.TagHeader
import gbl.tag.Tag
import gbl.tag.TagWithHeader

data class GblProgLzma(
    override val tagHeader: TagHeader,
    override val tagType: GblType,
    override val tagData: ByteArray
): Tag, TagWithHeader {
    override fun copy(): Tag {
        return GblProgLzma(
            tagHeader = tagHeader,
            tagType = tagType,
            tagData = arrayOf<Byte>().toByteArray()
        )
    }
}

File: python-library/tag/type/gbl_prog_lzma.py
from dataclasses import dataclass
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from ..gbl_type import GblType
    from ..tag_header import TagHeader
    from ..tag import Tag


@dataclass
class GblProgLzma:
    """GBL Program LZMA compressed tag"""

    tag_header: 'TagHeader'
    tag_data: bytes  # ByteArray -> bytes

    @property
    def tag_type(self) -> 'GblType':
        """Get tag type"""
        from ..gbl_type import GblType
        return GblType.PROG_LZMA

    def copy(self) -> 'Tag':
        """Create a copy of the tag"""
        return GblProgLzma(
            tag_header=self.tag_header,
            tag_data=bytes()  # Empty byte array як у Kotlin
        )